src/NewCommand.php

```diff
@@ -792,15 +783,29 @@ protected function canResolveHostname($hostname)
      */
     protected function isParked(string $directory)
     {
-        foreach (['herd', 'valet'] as $tool) {
-            $process = new Process([$tool, 'paths', '-v']);
+        $output = $this->runOnValetOrHerd('paths');
+
+        return $output !== false ? in_array(dirname($directory), json_decode($output)) : false;
+    }
 
-            $process->run();
+    /**
+     * Runs the given command on "herd" or "valet" cli.
+     *
+     * @param  string  $command
+     * @return string|bool
+     */
+    protected function runOnValetOrHerd(string $command)
+    {
+        foreach (['herd', 'valet'] as $tool) {
+            $process = new Process([$tool, $command, '-v']);
 
-            if ($process->isSuccessful()) {
-                $output = json_decode(trim($process->getOutput()));
+            try {
+                $process->run();
 
-                return in_array(dirname($directory), $output);
+                if ($process->isSuccessful()) {
+                    return trim($process->getOutput());
+                }
+            } catch (ProcessStartFailedException) {
             }
```

Comment:
Technically this is returning `null` and not false in case neither `herd` nor `valet` are present, isn't it?

In `isParked´ you're doing a strict compare with `false`, which might be an issue in such case.

Reply:
I don't think it will return `null`. If herd or valet is present I except the output to be correct.

I can increase the checking on the `json_decode` but not sure if is needed.

What do you think @driesvints?

Reply:
@xiCO2k but what if neither is present like @Anticom says. What will happen then?

Reply:
that will throw the `ProcessStartFailedException` and we will ignore it and try `valet` is the same happens we will ignore and then after the loop we will return `false` as before.

Reply:
Sorry guys, I missed it since it wasn't shown in the diff. My bad!